sc/shared/src/main/scala/sigma/compiler/ir/Base.scala

```diff
@@ -203,10 +204,11 @@ abstract class Base { thisIR: IRContext =>
     }
   }
 
-  /** Logical AND between two pattern matches of the save value `x`.
-    * Can be used to construct patterns like `case P1 && P2 => ...` */
-  object && {
-    def unapply[T](x: T): Option[(T,T)] = Some((x, x))
+  /**
+    * Def done in order to carry on DeserializeContext through stages of compilation intact
+    */
+  case class DeserializeContextDef[V <: SType](d: DeserializeContext[V], e: Elem[V]) extends Def[V] {
+    override def resultType: Elem[V] = e
```

Comment:
There is a problem with typing here. In a type like `Elem[V]`, the V should be a type of data value (Int, Long, Boolean, Coll, GroupElement). However, the `V <: SType` part suggests `V` to be SInt, SLong, SBoolean, etc.
The compiler is not complaining likely because of `.asInstanceOf[]` being used.
It should be `..., e: Elem[V#WrappedType]) extends Def[V#WrappedType] ...`